src/Storages/MergeTree/ReplicatedMergeTreePartCheckThread.cpp

```diff
@@ -194,7 +195,7 @@ void ReplicatedMergeTreePartCheckThread::searchForMissingPartAndFetchIfPossible(
         if (!storage.queue.remove(zookeeper, part_name))
         {
             /// The part was not in our queue. Why did it happen?
-            LOG_ERROR(log, "Missing part {} is not in our queue.", part_name);
+            throw Exception(ErrorCodes::LOGICAL_ERROR, "Missing part {} is not in our queue.", part_name);
```

Comment:
Do we this exception it in release build? It will stuck the queue.

Reply:
Yes, possibly too dangerous.

Reply:
However, this exception will be thrown in the checker thread and it will not affect the queue.